Update FiltersSummary.md

ginalee-dotcom opened the pull request:
Update according to https://microsofthealth.visualstudio.com/Health/_git/conversion-template-generation?path=%2Fdoc%2FCDA.md&version=GBpersonal%2Fboywu%2Fcda-document&anchor=new-filters&_a=preview

BoyaWu10 commented on docs/FiltersSummary.md:
```diff
@@ -40,6 +54,14 @@ If these filters do not meet your needs, you can also write your own filters.
 |-|-|-|
 | add_hyphens_date | Adds hyphens to a date without hyphens | `{{ PID.7.Value \| add_hyphens_date }}` |
 | format_as_date_time | Convert an YYYYMMDDHHmmssSSS string, e.g. 20040629175400000 to dateTime format, e.g. 2004-06-29T17:54:00.000z | `{{ PID.29.Value \| format_as_date_time }}` |
+| now | Provides current UTC time in "yyyy-MM-ddTHH:mm:ss.FFFZ" format | `{{ "" \| now }}` |
```
These two filters have been refined to be more powerful in recent updates (PR #222, etc). Could you please help to update the contents for these two filters?

| format_as_date_time | Convert an YYYYMMDDHHmmssSSS string, e.g. 20040629175400000 to dateTime format, e.g. 2004-06-29T17:54:00.000z. A parameter could be set to handle time zone with "preserve", "utc" or "local". The default method is "local" | `{{ PID.29.Value \| format_as_date_time: 'utc' }}` |
| now | Provides current time in a specific format. The default format is "yyyy-MM-ddTHH:mm:ss.FFFZ" | `{{ '' \| now: 'dddd, dd MMMM yyyy HH:mm:ss' }}` |

hongyh13 commented on docs/FiltersSummary.md:
```diff
@@ -16,13 +16,27 @@ If these filters do not meet your needs, you can also write your own filters.
+| base64_decode | Returns base64 decoded string | `{{ encodedData \| base64_decode }}` |
+| sha1_hash | Returns SHA1 hash (in hex) of given string | `{{ inputData \| sha1_hash }}` |
+| gzip | Returns compressed string | `{{ uncompressedData \| gzip }}` |
+| gunzip | Returns decompressed string | `{{ compressedData \| gunzip }}` |
```
`gunzip` is changed to `gunzip_base64_string` now.

ginalee-dotcom commented on docs/FiltersSummary.md:
```diff
@@ -40,6 +54,14 @@ If these filters do not meet your needs, you can also write your own filters.
+| concat | Returns the concatenation of provided arrays | `{% assign ethnicCodes = ethnicCodes1 \| concat: ethnicCodes2 -%}` |
+| batch_render | Render every entry in a collection with a snippet and a variable name set in snippet | `{{ firstSections.2_16_840_1_113883_10_20_22_2_5_1.entry \| to_array \| batch_render: 'Entry/Problem/entry', 'entry' }}` |
 
 ### Miscellaneous filters
```
@hongyh13 What do I put for `evaluate` and `generate_id_input` Description and Syntax? Is there somewhere I can find the info?

ginalee-dotcom replied:
@BoyaWu10 Do you know where I can find these filters? Thanks!

BoyaWu10 replied:
@ginalee-dotcom Sorry for missing these two.

For `generate_id_input`, we can add it in `Miscellaneous filters`:
| generate_id_input | Generates an input string for generate_uuid with 1) the resource type, 2) whether a base ID is required, 3) the base ID (optional) | `{{ identifiers \| generate_id_input: 'Observation', false, baseId \| generate_uuid }}` |

For `evaluate`, we may need a new section called `Tags` at the same level of `Filters` (`Tag` uses {% tag -%} while `Filters` uses `{{ data | filter_name }}`.):

By default, Liquid provides a set of standard [tags](https://github.com/Shopify/liquid/wiki/Liquid-for-Designers#tags) to assist template creation. Besides these tags, FHIR Converter also provides some other tags that are useful in conversion, which are listed below. If these tags do not meet your needs, you can also write your own tags.

| Tag | Description | Syntax |
| evaluate | Evaluates an ID with an ID generation template and input data | `{% evaluate patientId using 'Utils/GenerateId' obj: msg.ClinicalDocument.recordTarget.patientRole -%}` |

BoyaWu10 commented on docs/FiltersSummary.md:
```diff
@@ -39,7 +53,15 @@ If these filters do not meet your needs, you can also write your own filters.
+| format_as_date_time | Convert an YYYYMMDDHHmmssSSS string, e.g. 20040629175400000 to dateTime format, e.g. 2004-06-29T17:54:00.000z. A parameter could be set to handle time zone with "preserve", "utc" or "local". The default method is "local" | {{ PID.29.Value \| format_as_date_time: 'utc' }} |
+| now | Provides current time in a specific format. The default format is "yyyy-MM-ddTHH:mm:ss.FFFZ" | {{ '' \| now: 'dddd, dd MMMM yyyy HH:mm:ss' }} |
+
+#### Collection filters
```
It seems collection filters could be of the same level with other filters (using `###`).

hongyh13 commented on docs/Filters-and-Tags.md:
```diff
@@ -39,10 +54,26 @@ If these filters do not meet your needs, you can also write your own filters.
+By default, Liquid provides a set of standard [tags](https://github.com/Shopify/liquid/wiki/Liquid-for-Designers#tags) to assist template creation. Besides these tags, FHIR Converter also provides some other tags that are useful in conversion, which are listed below. If these tags do not meet your needs, you can also write your own tags.
+
+| Tag | Description | Syntax |
+| evaluate | Evaluates an ID with an ID generation template and input data | `{% evaluate patientId using 'Utils/GenerateId' obj: msg.ClinicalDocument.recordTarget.patientRole -%}` |
```
There seems some format issue in the tag table, it couldn't display normally as a table.

BoyaWu10 replied:
It looks that we are missing "|-|-|-|" between the title line and contents.

ginalee-dotcom replied:
Ah sorry I missed that :) Fixed!

BoyaWu10 commented:
Since we have updated the title of the document, we may also need to update the link in README.md: https://github.com/microsoft/FHIR-Converter#reference-documentation.

ginalee-dotcom commented:
> Since we have updated the title of the document, we may also need to update the link in README.md: https://github.com/microsoft/FHIR-Converter#reference-documentation.

Yup! Updated in the README PR.

BoyaWu10 commented on docs/Filters-and-Tags.md:
```diff
@@ -16,13 +17,27 @@ If these filters do not meet your needs, you can also write your own filters.
+| base64_decode | Returns base64 decoded string | `{{ encodedData \| base64_decode }}` |
+| sha1_hash | Returns SHA1 hash (in hex) of given string | `{{ inputData \| sha1_hash }}` |
+| gzip | Returns compressed string | `{{ uncompressedData \| gzip }}` |
+| gunzip_base64_string | Returns decompressed string | `{{ compressedData \| gunzip }}` |
```
The example syntax in the third column is still `gunzip`. We can updated it to `gunzip_base64_string` as well.

BoyaWu10 commented on docs/Filters-and-Tags.md:
```diff
@@ -16,13 +17,27 @@ If these filters do not meet your needs, you can also write your own filters.
+| get_first_ccda_sections | Returns first instance (non-alphanumeric chars replace by '_' in name) of the sections | `{% assign firstSections = msg \| get_first_cda_sections: 'Problems' -%}` |
+| get_ccda_section_lists | Returns instance list (non-alphanumeric chars replace by '_' in name) for the given sections | `{% assign sections = msg \| get_cda_section_lists: 'Problems' -%}` |
+| get_first_ccda_sections_by_template_id | Returns first instance (non-alphanumeric chars replace by '_' in name) of the sections by template id | `{% assign firstSections = msg \| get_first_cda_sections_by_template_id: '2.16.840.1.113883.10.20.22.2.5.1' -%}` |
+
```
The example syntax in the third columns still uses `cda`. We can update them to `ccda` as well for these three filters.

BoyaWu10 commented:
Hi @ginalee-dotcom, could you please update the document to `main` branch? The PR looks good and I try to merge the PR, but find out that it was based on `dotliquid` branch.

ginalee-dotcom commented:
Closing and reopening a new PR.
